### PR TITLE
Revamp

### PR DESCRIPTION
**Pull request by sandeeprao698**
Hi Mike,
This PR contains
- UI for home Page
- API to get transactions of last 24 hrs
- D3 js graph integrated for transactions of last 24 hrs
Known issue
- Graph does not have grid in background. we are trying to find a solution for same and we will update it as soon as we can.

### CODE REVIEW

File: src/components/Home/Banner/index.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Title } from './style';
+import { Text } from './style';
+import { Title2 } from './style';
+import { BottomDiv } from './style';
+import { Holder } from './style';
```
Comment (dariuszsobkowicz) [inline src/components/Home/Banner/index.tsx]: i suggest use single import

File: src/components/Home/Banner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import DownArrow from '../../../assets/icons/down-arrow.png';
+
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+
+
+const useStyles = makeStyles({
```
Comment (dariuszsobkowicz) [inline src/components/Home/Banner/index.tsx]: can be moved to styles

File: src/components/Home/Banner/index.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        outline: 'none !important',
+        border: '1px solid #FF7477;',
+        background: '#FF7477 !important',
+        color: '#fff',
+        width: 'max-content',
+        alignSelf: 'center',
+    },
+});
+
+const Banner = (props: any) => {
```
Comment (dariuszsobkowicz) [inline src/components/Home/Banner/index.tsx:15]: looks like `props` not in use

File: src/components/Home/Banner/style.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import styled from 'styled-components';
```
Comment (dariuszsobkowicz) [inline src/components/Home/Banner/style.tsx:1]: here can be used `import { styled } from '@material-ui/core/styles';`

File: src/containers/Homepage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+import { ContentHolder } from './style';
+import { Paragraph } from './style';
+import { ImageHolder } from './style';
+import { WhiteBg } from './style';
+import { SectionTitle } from './style';
+import { Footer } from './style';
+import { ListItem } from './style';
+import { ListContainer, styles } from './style';
+import { BannerHolder } from './style';
+import { MapHolder } from './style';
```
Comment (dariuszsobkowicz) [inline src/containers/Homepage/index.tsx]: same ^^^

File: src/containers/Homepage/index.tsx [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+                <WhiteBg>
+                <Container style={{marginBottom:'100px'}} maxWidth="lg">
+                    <Padding>
+                        <Grid container justify="center" alignItems="center" spacing={0}>
+                            <Grid item xs={6}>
+                                <SectionTitle style={{margin:0,justifyContent:'flex-start'}}>
+                                    Start exploring blockchain data
+                                </SectionTitle>
+                            </Grid>
+                            <Grid style={{textAlign: 'right'}} item xs={6}>
```
Comment (dariuszsobkowicz) [inline src/containers/Homepage/index.tsx]: do we need inline styles?

File: src/containers/Homepage/index.tsx [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+    }
+
+    async componentDidMount() {
+        this.fetchHourlyTransactions();
+    }
+
+    async fetchHourlyTransactions() {
+        const { loadHourlyTransactions } = this.props;
+
+        let prevDate = new Date();
```
Comment (dariuszsobkowicz) [inline src/containers/Homepage/index.tsx]: can be `const`

File: src/containers/Homepage/index.tsx [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+        prevDate.setDate(prevDate.getDate() - 1);
+
+        await loadHourlyTransactions(prevDate.getTime());
+        let {data, label, timestamps } = await this.formatHourlyTransactions(this.props.hourlytransactions, prevDate.getTime());
+
+        this.generateHourlyTransactionsGraph(data, label, timestamps);
+    }
+
+    async formatHourlyTransactions(transactionsList: Transactions[], date: number) {
+        let label: Array<Date> = [],
```
Comment (dariuszsobkowicz) [inline src/containers/Homepage/index.tsx]: i think can be `const`, can't see it is overwritten

File: src/containers/Homepage/types.tsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    
+    classes: any;
+    hourlytransactions: Transactions[];
+    isError: boolean;
+    message: string;
+    loadHourlyTransactions: (date: number) => void;
+    initMessage: () => void;
+}
+
+export interface States {
```
Comment (dariuszsobkowicz) [inline src/containers/Homepage/types.tsx:29]: if stays empty i don't think is necessary

File: src/components/Home/Banner/index.tsx (hunk 1 above)
Comment (sandeeprao698) [reply src/components/Home/Banner/index.tsx]: Fixed

File: src/components/Home/Banner/index.tsx (hunk 2 above)
Comment (sandeeprao698) [reply src/components/Home/Banner/index.tsx]: fixed

File: src/components/Home/Banner/index.tsx (hunk 3 above)
Comment (sandeeprao698) [reply src/components/Home/Banner/index.tsx:15]: fixed

File: src/components/Home/Banner/style.tsx (hunk 4 above)
Comment (sandeeprao698) [reply src/components/Home/Banner/style.tsx:1]: if we do this, styled.h1 gives error`Property 'h1' does not exist on type '<Component extends ElementType<any>>(Component: Component) => ComponentCreator<Component>'`

File: src/containers/Homepage/index.tsx (hunk 5 above)
Comment (sandeeprao698) [reply src/containers/Homepage/index.tsx]: Fixed

File: src/containers/Homepage/index.tsx (hunk 7 above)
Comment (sandeeprao698) [reply src/containers/Homepage/index.tsx]: Fixed

File: src/containers/Homepage/index.tsx (hunk 8 above)
Comment (sandeeprao698) [reply src/containers/Homepage/index.tsx]: fixed

File: src/containers/Homepage/types.tsx (hunk 9 above)
Comment (sandeeprao698) [reply src/containers/Homepage/types.tsx:29]: Fixed

File: src/containers/Homepage/index.tsx (hunk 6 above)
Comment (sandeeprao698) [reply src/containers/Homepage/index.tsx]: fixed

**Review by anonymoussprocket — APPROVED** [review]

**Review by vishakh — APPROVED** [review]

**Pull request merged**